Lesson8/main.js

```diff
@@ -0,0 +1,22 @@
+const addLoading = () =>{
+  const loading = document.getElementById("js-loading");
+  const loadingImg = document.createElement('img');
+  loadingImg.src = 'loading.gif';
+  loadingImg.id = "loading-img"
+  loading.appendChild(loadingImg)
+}
+
+const promiseChecker = new Promise((resolve, reject) => {
+  addLoading();
+  setTimeout(() => {
+    reject("The promise was rejected!");
+  }, 3000);
+});
+
+promiseChecker.then(
+  (value) => {
+    console.log('OK', value)
+})
+.catch((error) =>{
+  console.error(error)
+});
```

Comment:
The code is correct.

Either one is fine, but in the case of this lesson,
 I think the following will be the correct answer, since you asked me to process it with then.
```suggestion
promiseChecker.then(
  (value) => {
    console.log('OK', value);
  },
  (error) => {
    console.error(error);
  }
);
```